core/metrics/src/main/scala/org/typelevel/otel4s/metrics/ObservableInstrumentBuilder.scala

```diff
@@ -45,5 +47,7 @@ trait ObservableInstrumentBuilder[F[_], A] {
 
   /** Creates an instrument with the given `unit` and `description` (if any).
     */
```

Comment:
It's worth mentioning how the callback should behave. (mostly taken from OpenTelemetry Java doc)

```suggestion
  /** Creates an instrument with the given callback, using `unit` and
    * `description` (if any).
    *
    * The callback will be called when the instrument is being observed.
    *
    * The callback is expected to abide by the following restrictions:
    *   - Short-living and (ideally) non-blocking
    *   - Run in a finite amount of time
    *   - Safe to call repeatedly, across multiple threads
    *
    * @param cb
    *   The callback which observes measurements when invoked
    */
```